src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts

```diff
@@ -18,7 +18,16 @@ export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWiza
 
         let result: FullFunctionAppStack | undefined;
         while (true) {
-            result = (await context.ui.showQuickPick(this.getPicks(context), { placeHolder, enableGrouping: true })).data;
+            const options: AgentQuickPickOptions = {
```

Comment:
I would prefer if the rest of the extension code basically had no idea about the `agentMetaData` stuff. Would we pass it in as an option instead when you invoke it here? https://github.com/microsoft/vscode-azurefunctions/pull/3979/files#diff-ed50eea82ea7bad017f14ed53416d4b50d441da3ce4240af90885e92c175a1b8R56

Reply:
> I would prefer if the rest of the extension code basically had no idea about the agentMetaData stuff. Would we pass it in as an option instead when you invoke it here?

Every call to a UI prompt needs to have various pieces of agent metadata added on. I can't think of a way to not accomplish that without sprinkling the agent metadata at each spot a UI prompt is done. Can you be more specific about what alternative ideas you have?

Reply:
Here specifically, I was originally think we could have had `options.agentMetadata` get passed into the constructor of `FunctionAppStackStep`. But that's just moving the agent code somewhere else still in the extension code base which isn't really alienating the agent code then...

And since each quick pick item itself needs to also have `agentMetadata` there probably isn't anyway to get around it. 

Oh well, no biggie. Resolving the conversation-- if I have any ideas later, I'll bring it up then.